packages/extensions/openapi-to-typespec/src/transforms/transform-arm-resources.ts

```diff
@@ -704,8 +711,24 @@ function getTspOperations(armSchema: ArmResourceSchema): [TspArmResourceOperatio
   return [tspOperations, normalOperations];
 }
 
-function getOperationName(name: string): string {
-  return _.lowerFirst(_.last(name.split("_")));
+const nameCache: { [resourceName: string]: Set<string> } = {};
+
+function getOperationName(resourceName: string, operationId: string): string {
+  let operationName = _.lowerFirst(_.last(operationId.split("_")));
+  if (resourceName in nameCache) {
```

Comment:
This part seems strange to me, with the cache in place, we still need to do some calculation from the cache out value.

Reply:
Yes, if the name is in cache, it means this name is already taken, so we need another name.